packages/generator-ds/src/component/index.ts

```diff
@@ -70,12 +70,14 @@ export default class ComponentGenerator extends Generator<ComponentGeneratorOpti
     const rawPath = super.destinationPath(...inPath);
     const dirName = path.dirname(rawPath);
 
+    // If the path is an ancestor of the cwd, we don't want to PascalCase the parent directory, as this would be an incorrect absolute filepath
+    // If the path is a descendant of the cwd, we want to PascalCase the parent directory to enforce React component naming conventions.
+    const componentParentDir = paths.isAncestor(this.env.cwd, rawPath)
+      ? path.basename(dirName)
+      : casing.toPascalCase(path.basename(dirName));
```

Comment:
This can be simplified by only appending ../pascalcase() to the path if the path is an ancestor instead of having a ternary, I'll tweak that when I'm next in the office :)